src/main/java/org/dbos/apiary/mongo/MongoContext.java

```diff
@@ -49,11 +57,17 @@ public FunctionOutput apiaryCallFunction(String name, Object... inputs) throws E
         return null;
     }
 
-    public void insertOne(String collectionName, Document document, String id) {
+    public void insertOne(String collectionName, Document document, String id) throws PSQLException {
         if (!ApiaryConfig.XDBTransactions) {
             database.getCollection(collectionName).insertOne(document);
             return;
         }
+        lockManager.putIfAbsent(collectionName, new ConcurrentHashMap<>());
+        lockManager.get(collectionName).putIfAbsent(id, new AtomicBoolean(false));
+        boolean available = lockManager.get(collectionName).get(id).compareAndSet(false, true);
+        if (!available) {
```

Comment:
I think usually read committed isolation shouldn't throw serialization failure -- concurrent transactions could modify tuples/records in their own snapshot. Does it mean that the "read committed" level here in secondary storage is actually stronger than normal?